lightning/src/ln/channelmanager.rs

```diff
@@ -6122,14 +6122,50 @@ impl<Signer: Sign, M: Deref , T: Deref , K: Deref , F: Deref , L: Deref >
 	}
 
 	fn provided_node_features(&self) -> NodeFeatures {
-		NodeFeatures::known()
+		provided_node_features()
 	}
 
 	fn provided_init_features(&self, _their_init_features: &PublicKey) -> InitFeatures {
-		InitFeatures::known_channel_features()
+		provided_init_features()
 	}
 }
 
+/// Fetches the set of [`NodeFeature`] flags which are provided by or required by
+/// [`ChannelManager`].
+pub fn provided_node_features() -> NodeFeatures {
+	provided_init_features().to_context()
+}
+
+/// Fetches the set of [`InvoiceFeature`] flags which are provided by or required by
+/// [`ChannelManager`].
+pub fn provided_invoice_features() -> InvoiceFeatures {
+	provided_init_features().to_context()
+}
+
+/// Fetches the set of [`ChannelFeature`] flags which are provided by or required by
+/// [`ChannelManager`].
+pub fn provided_channel_features() -> ChannelFeatures {
+	provided_init_features().to_context()
+}
```

Comment:
Thoughts on having one method?

```rust
pub fn provided_features<T: From<InitFeatures>>() -> T { ... }
```

Not sure if having it based on `InitFeatures` would be weird.